Sources/BraintreeCoreSwift/BTAnalyticsSession.swift

```diff
@@ -0,0 +1,14 @@
+import Foundation
+
+/// Encapsulates analytics events for a given session
+struct BTAnalyticsSession {
+
+    let sessionID: String
+    let source: String
+    let integration: String
+
+    var events: [BTAnalyticsEvent] = []
```

Comment:
Nitpick comment - but I was looking at Northstar and we don't really have consistency around when new structs (data model types) get new files, vs get added to existing files.

I feel like a good rule we can follow is - if the type is only used in one other model, it can live in the same file (so in this case, moving `BTAnalyticsEvent` definition to live in this file). If the type is used within various models, it should live in it's own file.

What are your thoughts on this? We can come up with a standard and include it in our Swift Style Guide.

Reply:
Yeah, I don't feel strongly one way or another. Maybe it makes sense to have a line length requirement where we put them in the same file vs not? I feel like if a model is only used in once place but is like 100 lines long it could make more sense to separate out into its own file, but also that get's into a grey area.

I feel like it is nice when looking at something like BTAnalyticsSession to not need to scroll past ~30 lines of model structs and instead can just hop right into the implementation, but also can see that having a dedicated file for 2 pretty small structs makes the overall project feel busier. Maybe it makes sense to have both `BTAnalyticsEvent` and `BTAnalyticsSession` in one file called `BTAnalyticsSession+Models` (not sure if that's a great idea just thinking out loud).

I can go both ways on it, so whatever we feel is best we should add to the style guide and implement here! It would be great to have this consistent across both SDKs. 😄

Reply:
Those are good points!

At this point, I think I would prefer slightly longer model files over more noise/files to sift through in the Xcode project. We could always change our preference if we see a pattern of model files getting huge, but I imagine it would be unlikely.

@mattwylder - any preference?

Reply:
Updated this in this commit: https://github.com/braintree/braintree_ios/pull/887/commits/a7e45b96a4083fe3a114b266fb812d4044ac7b99